src/tts_text/phoneme_covering.py

```diff
@@ -0,0 +1,194 @@
+"""Module for creating a phoneme covering set."""
+
+import json
+import re
+import logging
+from collections import Counter
+from pathlib import Path
+from tqdm.auto import tqdm
+
+from datasets import load_dataset, Dataset
+from omegaconf import DictConfig
+
+logger = logging.getLogger(__name__)
+
+
+def sort_wiki_by_phoneme_occurence(cfg: DictConfig) -> Dataset:
+    """Sort the wiki dataset by phoneme occurence.
+
+    Args:
+        cfg: The Hydra configuration object.
+
+    Returns:
+        The sorted dataset.
+
+    Raises:
+        ValueError: If sort_by is not one of "da", "en" or "all".
+    """
+    # The `wiki40b` dataset is a small dataset so we can load it all into memory
+    # instead of streaming it.
+    dataset = load_dataset(
+        "wiki40b",
+        "da",
+        split="train",
+        beam_runner="DirectRunner",
+    )
+
+    with Path(cfg.dirs.phoneme_dir).open() as f:
+        phonemes = json.load(f)
+
+    # Count phonemes in articles
+    dataset = dataset.map(
+        lambda x: count_occurences(x, phonemes),
+    )
+    sort_by = cfg.phoneme_sort_strategy
+    if sort_by == "da":
+        dataset = dataset.sort("da_unique_phonemes_count", reverse=True)
+    elif sort_by == "en":
+        dataset = dataset.sort("en_unique_phonemes_count", reverse=True)
+    elif sort_by == "all":
+        dataset = dataset.sort("all_unique_phonemes_count", reverse=True)
+    else:
+        raise ValueError(f"sort_by must be one of 'da', 'en' or 'all', got {sort_by}")
+
+    return dataset
+
+
+def count_occurences(document: dict, phonemes: dict) -> dict:
```

Comment:
Again, difficult to understand what this does based on the name. Maybe `count_phoneme_occurences`?